CMakeFuncs.cmake

```diff
@@ -0,0 +1,627 @@
+# This is a set of common functions and settings for rippled
```

Comment:
As a first approximation of a guess:
1. Because that's just a subdirectory description. A lot of the complexity is hidden in functions defined in https://github.com/vinniefalco/NuDB/blob/master/CMakeLists.txt
2. Rippled's CMakeLists.txt came first, so it's rougher.
3. Rippled has to support a lot more build options and combinations that NuDB doesn't appear to support, or supports singly (eg. unity/nounity, gcc/clang, static, etc).

Not making excuses, but complexity does breed complexity.